### PR TITLE
[37026] Add new project form with template selection

### PR DESCRIPTION
**Pull request by oliverguenther**
https://community.openproject.org/wp/37026

- [x] Switch to copy form when template selected
  - [x] Hide the meta fields from new form for now
- [x] On submission, switch to the status modal

### CODE REVIEW

File: frontend/src/app/modules/projects/components/new-project/new-project.component.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  }
+
+  ngOnInit():void {
+    this.resourcePath = this.pathHelperService.projectsPath();
+
+    if (this.uIRouterGlobals.params.parent_id) {
+      this.setParentAsPayload(this.uIRouterGlobals.params.parent_id);
+    }
+
+    this.dynamicFieldsSettingsPipe = (dynamicFieldsSettings) => {
```
Comment (b12f) [inline frontend/src/app/modules/projects/components/new-project/new-project.component.ts]: For readability, I'd prefer if this were a class method, and then bound in the template with `bind(this)`

File: frontend/src/app/modules/projects/components/new-project/new-project.component.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<form
+  class="op-form"
+  [formGroup]="templateForm"
+>
+  <op-form-field
+    [label]="text.use_template"
+    data-field-name="use_template"
+  >
+    <ng-select
+      [virtualScroll]="true"
```
Comment (b12f) [inline frontend/src/app/modules/projects/components/new-project/new-project.component.html]: Not sure if we need it here

**Review by b12f — APPROVED** [review]
LGTM, thought two points;

1. You might want to wait for https://github.com/opf/openproject/pull/9215 and add the right attribute / scope to form fields right away
2. I'd rename `data-field-name` to `data-qa-name` or something similar to make its purpose more clear.

File: frontend/src/app/modules/projects/components/new-project/new-project.component.ts (hunk 1 above)
Comment (oliverguenther) [reply frontend/src/app/modules/projects/components/new-project/new-project.component.ts]: I agree. Most of this is actually from the upstream PR, and I don't find it readable at all. It still needs to be extracted into more than one method, ideally with some common helpers.

For now I did extract it into a  method though

**Pull request merged**